README.md

```diff
@@ -1,2 +1,29 @@
-# openmrs-frontend-spa
-The frontend SPA for OpenMRS code
+# openmrs-module-spa
+
+This project is an OpenMRS java module that makes it possible for Tomcat to serve an SPA instead of expecting all routes
+to be handled by server side JSPs/GSPs.
+
+It is part of the microfrotends architecture that is explained at https://github.com/openmrs/openmrs-rfc-frontend.
+
+## Development instructions
+1. Clone this repository.
+2. `mvn clean install`
+3. Copy `omod/target/spa-1.0.0-SNAPSHOT.jar` to the `~/.OpenMRS/modules` directory. On mac/linux, this is `cp omod/target/spa-1.0.0-SNAPSHOT.omod ~/.OpenMRS/modules`
+4. Add the following to [the web.xml file in openmrs-core](https://github.com/openmrs/openmrs-core/blob/master/webapp/src/main/webapp/WEB-INF/web.xml).
+   This step is one we need to get rid of or work around. My hope is that we can get openmrs-core to respect the
+   [web-fragment.xml](/omod/src/main/resources/META-INF/web-fragment.xml) file in this repository, but I have not succeeded
+   in doing so yet, so we've got this manual step.
+```xml
+<servlet>
+    <servlet-name>default</servlet-name>
+    <servlet-class>org.openmrs.module.spa.web.SinglePageApplicationServlet</servlet-class>
+</servlet>
+
+<servlet-mapping>
+    <servlet-name>default</servlet-name>
+    <url-pattern>/</url-pattern>
+</servlet-mapping>
```

Comment:
@mseaton  @brandones  and I were talking about this today... in some ways it would be good not to touch core, but, in this case, since this is designed be such a fundamental feature of OpenMRS going forward, I'd be in support of changing Core if need be... from a PIH EMR perspective, we have finally gotten ourselves on a recent version of OpenMRS Core, so upgrading should be much work.  Would be good to check with AMPATH and Bahmni.

Also, if it turns out we are going to need to modify core regardless, it would likely make sense to *not* create a new module but instead just incorporate all the changes in this module into core.

So, in summary, if it's easy enough to set this all up in a module, let's start with that, but if we end up having to go through a lot of hoops to get things working, we should consider just modifying core.

Reply:
👍 yeah I agree. I was thinking that I'd start it out as a separate module because that seemed less intrusive or likely to break existing code, but then as we grow more confident in it we can decide whether to just bring it into core.